packages/editor/src/components/tabs/name/NameTab.tsx

```diff
@@ -13,24 +13,45 @@ function useNameTabValidation(): InscriptionValidation[] {
 
 export function useNameTab(options?: { hideTags?: boolean }): TabProps {
   const validation = useNameTabValidation();
-  const { data } = useDataContext();
-  const tabState = useTabState(['', '', [], []], [data.name, data.description, data.docs, data.tags], validation);
-  return { name: 'Name', state: tabState, content: <NameTab hideTags={options?.hideTags} /> };
+  const { data, initData, resetData } = useNameData();
+  const currentData = [data.name, data.description, data.docs, data.tags];
+  const tabState = useTabState(['', '', [], []], currentData, validation);
+  const tabDirty = useTabDirty([initData.name, initData.description, initData.docs, initData.tags], currentData);
+  return {
+    name: 'Name',
+    state: tabState,
+    reset: { dirty: tabDirty, action: () => resetData() },
+    content: <NameTab hideTags={options?.hideTags} />
+  };
 }
 
 const NameTab = (props: { hideTags?: boolean }) => {
-  const { data, updateName, updateDescription, updateDocs, updateTags } = useNameData();
+  const { data, initData, updateName, updateDescription, updateDocs, updateTags } = useNameData();
   const [nameValidation, descriptionValidation] = useNameTabValidation();
 
   return (
     <>
-      <Fieldset label='Display name' htmlFor='displayName' controls={[ResetControl]} message={nameValidation}>
+      <Fieldset
+        label='Display name'
+        htmlFor='displayName'
+        reset={{ data: data.name, initData: initData.name, resetData: () => updateName(initData.name) }}
+        message={nameValidation}
+      >
         <Textarea rows={1} id='displayName' value={data.name} onChange={change => updateName(change)} />
       </Fieldset>
-      <Fieldset label='Description' htmlFor='description' message={descriptionValidation}>
+      <Fieldset
+        label='Description'
+        htmlFor='description'
+        reset={{ data: data.description, initData: initData.description, resetData: () => updateDescription(initData.description) }}
+        message={descriptionValidation}
+      >
         <Textarea rows={2} id='description' value={data.description} onChange={change => updateDescription(change)} />
       </Fieldset>
-      <Fieldset label='Means / Documents' htmlFor='documents'>
+      <Fieldset
+        label='Means / Documents'
+        htmlFor='documents'
+        reset={{ data: data.docs, initData: initData.docs, resetData: () => updateDocs(initData.docs) }}
```

Comment:
looks to me as if we sort of duplicate:
- the data.binding 
- the bound update method
... which raises the chance to forget things. 

wouldn't it  be cooler, if we could generalize this in the widget; rather than on the glue code? 
looks to me as if the 'widget' could hold it's init data and fire a 'onChange' for resetting the value.

Reply:
I know what you mean. But I'm not sure how I can achieve this...



**What is not possible:**
That we apply the data to the fieldset and this one gives it to it's children, because the Fieldset doesn't define its child itself.

**What can be possible:**
That the Fieldset defines a context where the Input (Textarea component) reads from. But I'm not sure if this really simplifies the code, and if this is a the common way to solve it...

Reply:
ok for me to keep it as is, it we don't think that the code really gets simpler. 
maybe we can keep it as a "polishing" task for upcoming stories. as we are currently mainly trying to adopt the UI concept.

Reply:
What we can do, is that the fieldset and the different widgets in in takes the same data object, e.g like the following:

```typescript
const NameTab = (props: { hideTags?: boolean }) => {
  const { data, initData, updateName, updateDescription, updateDocs, updateTags } = useNameData();
  const [nameValidation, descriptionValidation] = useNameTabValidation();

  const nameData = { data: data.name, initData: initData.name, updateData: updateName };
  const descriptionData = { data: data.description, initData: initData.description, updateData: updateDescription };
  const docsData = { data: data.docs, initData: initData.docs, updateData: updateDocs };

  return (
    <>
      <Fieldset label='Display name' htmlFor='displayName' data={nameData} message={nameValidation}>
        <Textarea rows={1} id='displayName' data={nameData} />
      </Fieldset>
      <Fieldset label='Description' htmlFor='description' data={descriptionData} message={descriptionValidation}>
        <Textarea rows={2} id='description' data={descriptionData} />
      </Fieldset>
      <Fieldset label='Means / Documents' data={docsData}>
        <DocumentTable data={docsData} />
      </Fieldset>
      {!props.hideTags && (
        <CollapsiblePart collapsibleLabel='Tags' defaultOpen={data.tags !== undefined && data.tags.length > 0}>
          <Tags tags={data.tags ?? []} onChange={updateTags} />
        </CollapsiblePart>
      )}
    </>
  );
};
```